content/technotes/2020-04-23-rmd-learnings/index.Rmd

```diff
@@ -0,0 +1,257 @@
+
+to appear -- as the result of a chunk producing a figure -- in the `.md` after knitting the `.Rmd`?
+
+I asked this question in the friendly French-speaking R Slack workspace[^grrr] and got an answer from both [Romain Lesur](https://github.com/RLesur) and [Christophe Dervieux](https://github.com/cderv): the solution was to use a [knitr plot hook](https://yihui.org/knitr/hooks/)!
+
+<!--html_preserve-->
+{{< figure src ="person-holding-purple-crochet-hook-and-white-yarn-3945638.jpg" alt = "Person holding a purple crochet hook and white yarn" link = "https://www.pexels.com/photo/person-holding-purple-crochet-hook-and-white-yarn-3945638/" caption = "Another type of hook. [Castorly Stock on Pexels](https://www.pexels.com/photo/person-holding-purple-crochet-hook-and-white-yarn-3945638/)." width = "300" class = "center" >}}
+<!--/html_preserve-->
+
+[knitr hooks](https://yihui.org/knitr/hooks/) are _"Customizable functions to run before / after a code chunk, tweak the output, and manipulate chunk options"_.
+
+In the setup chunk[^setup] of the `.Rmd` file, there should be this code
+
+```r
+knitr::knit_hooks$set(
+  plot = function(x, options) {
+    hugoopts <- options$hugoopts
+    paste0(
+      "{", "{<figure src=", # the original code is simpler
+      # but here I need to escape the shortcode!
+      '"', x, '" ',
+      if (!is.null(hugoopts)) {
+        glue::glue_collapse(
+          glue::glue('{names(hugoopts)}="{hugoopts}"'),
+          sep = " "
+        )
+      },
+      ">}}\n"
+    )
+  }
+)
+```
+
+that reads options from the chunk, and uses options from the `hugoopts` named list if it exists. 
+
+The chunk
+
+````markdown
+```{r chunkname, hugoopts=list(alt="alternative text please make it informative", title="title of the image", caption="this is what this image shows, write it here or in the paragraph after the image as you prefer", width=300)} `r ''`
+plot(1:10)
+```
+````
+
+produces
+
+<!--html_preserve-->
+{{</* figure src="chunkname-1.png" alt="alternative text please make it informative" title="title of the image" caption="this is what this image shows, write it here or in the paragraph after the image as you prefer" width="300" */>}}
+<!--/html_preserve-->
+
+in the `.md` file which is what we wanted.
+
+Now, a bit later in our website journey, I had a quite similar question: Hugo has nice [highlighting options for code fences](https://gohugo.io/content-management/syntax-highlighting/#highlighting-in-code-fences). How to make R Markdown show R source code with such options? This time there was no need to ask anyone, searching the internet for the _name_ of the right knitr hook was enough: our `.Rmd` has to feature a [knitr source hook](https://bookdown.org/yihui/rmarkdown-cookbook/output-hooks.html). More on that highlighting chapter another time, but in the meantime refer to [our standard `.Rmd`](https://github.com/ropensci/roweb2/blob/master/archetypes/Rmd/index.md).
```

Comment:
Would be cool to remember after the highlighting post is published, to come back here and add a little update sentence with a link that that post.

Reply:
Yes definitely!